velox/connectors/utils/StringFormatter.h

```diff
@@ -147,7 +161,18 @@ struct DefaultFormatter : public StringFormatter {
     } else if constexpr (std::is_same_v<T, StringView>) {
       ss << t.str();
     } else if constexpr (std::is_same_v<T, Timestamp>) {
-      ss << t.toString();
+      TimestampToStringOptions options;
+      uint64_t nanos = t.getNanos();
+      if (nanos % 1000 != 0) {
+        options.precision = TimestampPrecision::kNanoseconds;
```

Comment:
So the precision adapts to the timestamp value? This differs from Spark and Presto in the cast, which use fixed precision options.

Reply:
Do you mean that you are using a fixed precision option config to specify the timestamp precision? And In that way, all the timestamp in the table are same precision, what if a table have 2 different precision timestamp fields , such as in flink `tbl(a timestamp(3), b timestamp(6))` ?